[BUG] [JAVA] Schema Mapping with generics skips wrong path, fails with file error on windows

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?

##### Description

When I specify a *schema mapping* (aka `schemaMappings` map in a config.yaml), where the mapped value contains a java type with generic parameters (e.g. `Wrapper<Thing>`), then the code generation aborts with

```
Exception in thread "main" java.lang.RuntimeException: Could not process model 'ThingWrapper'.Please make sure that your schema is correct!
[...]
Caused by: java.nio.file.InvalidPathException: Illegal char <<> at index 53: .\src\main\java\org\openapitools\client\model\Wrapper<Thing>.java
```

at https://github.com/OpenAPITools/openapi-generator/blob/2abcb4f7f0aa67c126304e43602ae72a83f2dca7/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L491

##### openapi-generator version

At least since 7.4.0, still broken on master

##### OpenAPI declaration file content or url

https://gist.github.com/Xiphoseer/05c1b92d36ff3beef1e7e84fc476ccf9

##### Generation Details

- Java Codegen, but applies to all `DefaultCodegen` where `toModelFilename` resolves schemaMappings
- Uses `schemaMappings`

##### Steps to reproduce

- Save `api.yaml` and `config.yaml` in a directory
- Run `openapi-generator-cli generate -c .\config.yaml`

It's possible / likely this particular error only occurs on windows, Linux will let me create a file with `<` in the name just fine,
but my understanding is that just hides the underlying issue.

##### Related issues/PRs

None found

##### Suggest a fix

I'd suggest removing the entire loop marked as "HACK":
https://github.com/OpenAPITools/openapi-generator/blob/2abcb4f7f0aa67c126304e43602ae72a83f2dca7/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L488-L493

It doesn't quite do what it's supposed to (skip generation of model templated files for models that are schema mapped), because `toModelFilename` for Java calls `toModelName` which considers `schemaMappings` and thus acts on the *mapped model name* and file, not the file it would generate without the schema mapping. Additionally, the function where this happens seems to ensure anyway (using `continue` after the loop) that the schema will not be imported for generation.

This overload in `AbstractJavaCodegen` is what causes the issue, but it is a valid implementation since Java does require the class and file name to match:
https://github.com/OpenAPITools/openapi-generator/blob/2abcb4f7f0aa67c126304e43602ae72a83f2dca7/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java#L930-L933

The default implementation is not susceptible to this because it's stateless:
https://github.com/OpenAPITools/openapi-generator/blob/2abcb4f7f0aa67c126304e43602ae72a83f2dca7/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L1637-L1639

So far I did not reason about the other generators.

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java
@@ -469,14 +469,7 @@ public class DefaultGenerator implements Generator {
             try {
                 //don't generate models that have an import mapping
                 if (config.schemaMapping().containsKey(name)) {
-                    LOGGER.debug("Model {} not imported due to import mapping", name);
-
-                    for (String templateName : config.modelTemplateFiles().keySet()) {
-                        // HACK: Because this returns early, could lead to some invalid model reporting.
-                        String filename = config.modelFilename(templateName, name);
-                        Path path = java.nio.file.Paths.get(filename);
-                        this.templateProcessor.skip(path, "Skipped prior to model processing due to schema mapping.");
-                    }
+                    LOGGER.info("Model {} not generated due to schema mapping", name);
                     continue;
                 }
 

